packages/composites/proof-of-reserves/src/endpoint/reserves.ts

```diff
@@ -34,23 +34,48 @@ const inputParameters: InputParameters = {
     type: 'array',
     description: 'An array of addresses to get the balance from, when `protocol` is set to `list`',
   },
+  disableAddressValidation: {
+    required: false,
+    type: 'boolean',
+    description: 'Gives the option to disable address validation before the balances are fetched.',
+  },
+  disableDuplicateAddressFiltering: {
+    required: false,
+    type: 'boolean',
+    description:
+      'Gives the option to disabled the filtering of duplicate addresses in a request. ' +
+      'If this is set to `true` and a duplicate address is contained in the request, the balance of that address will be counted twice.',
+  },
 }
 
 export const execute: ExecuteWithConfig<Config> = async (input, context, config) => {
   const validator = new Validator(input, inputParameters, paramOptions)
-
   const jobRunID = validator.validated.jobRunID
   const protocol = validator.validated.data.protocol.toUpperCase()
   const indexer: Indexer = validator.validated.data.indexer.toUpperCase()
   const confirmations = validator.validated.data.confirmations
 
   const protocolOutput = await runProtocolAdapter(jobRunID, context, protocol, input.data, config)
+  const validatedInput = { ...protocolOutput }
+  if (
+    !validator.validated.data.disableAddressValidation ||
+    validator.validated.data.disableAddressValidation !== 'true'
+  ) {
+    validatedInput.result = validateAddresses(indexer, validatedInput.result)
+  }
+  if (
+    !validator.validated.data.disableDuplicateAddressFiltering ||
+    validator.validated.data.disableDuplicateAddressFiltering !== 'true'
+  ) {
+    validatedInput.result = filterDuplicates(validatedInput.result)
+  }
+  validatedInput.data.result = validatedInput.result
```

Comment:
I'm concerned with how big this logic is growing here. Let's extract it just like how we do it with the Reduce step: https://github.com/smartcontractkit/external-adapters-js/blob/85f6de8abad564af166a11c340b54fce1578469f/packages/composites/proof-of-reserves/src/utils/reduce.ts#L24-L51